contracts/src/v0.8/automation/chains/ScrollModule.sol

```diff
@@ -15,14 +19,22 @@ contract ScrollModule is ChainModuleBase {
   address private constant SCROLL_ORACLE_ADDR = 0x5300000000000000000000000000000000000002;
   IScrollL1GasPriceOracle private constant SCROLL_ORACLE = IScrollL1GasPriceOracle(SCROLL_ORACLE_ADDR);
 
+  /// @dev L1 fee coefficient can be applied to reduce possibly inflated gas cost
+  uint8 public s_l1FeeCoefficient = 100;
   uint256 private constant FIXED_GAS_OVERHEAD = 45_000;
   uint256 private constant PER_CALLDATA_BYTE_GAS_OVERHEAD = 170;
 
-  function getCurrentL1Fee() external view override returns (uint256) {
-    return SCROLL_ORACLE.getL1Fee(bytes.concat(msg.data, SCROLL_L1_FEE_DATA_PADDING));
+  constructor() ConfirmedOwner(msg.sender) {}
+
+  function getCurrentL1Fee(uint256 dataSize) external view override returns (uint256) {
```

Comment:
have we tested this function on scroll to see if it is charging reasonably?

Reply:
yes. i did some testing with a custom upkeep and this module. 
i think we can start with 100 as the coefficient and reduce it to a comfortable number